### PR TITLE
[DOC] Storage Considerations - New and modified content for ENTMQST-555

### PR DESCRIPTION
**Pull request by PaulRMellor**
### Type of change

Storage considerations 

- Documentation

### Description

_Please describe your pull request_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: documentation/book/con-considerations-for-data-storage.adoc [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+//
+// assembly-overview.adoc
+
+[id='considerations-for-data-storage-{context}']
+
+= Data storage considerations
+
+An efficient data storage infrastructure is essential to the optimal performance of {ProductName}.
+
+{ProductName} requires block storage and works well with cloud-based block storage solutions, including Amazon Elastic Block Store (EBS). The use of file storage is not recommended.
```
Comment (sjay72) [inline documentation/book/con-considerations-for-data-storage.adoc]:
```suggestion
{ProductName} requires block storage and is designed to work optimally with cloud-based block storage solutions, including Amazon Elastic Block Store (EBS). The use of file storage is not recommended.
```

File: documentation/book/assembly-storage.adoc [hunk 2]
```diff
@@ -10,7 +10,7 @@
 [id='assembly-storage-{context}']
 = Kafka and Zookeeper storage
 
-As stateful applications, Kafka and Zookeeper need to store data on disk. {ProductName} supports three different types of storage for this data: ephemeral, persistent, and JBOD (Just a Bunch of Disks) storage.
+As stateful applications, Kafka and Zookeeper need to store data on disk. {ProductName} supports three different types of storage for this data: ephemeral, persistent, and JBOD storage.
 
 NOTE: JBOD storage is only supported for Kafka, not for Zookeeper.
```
Comment (sjay72) [inline documentation/book/assembly-storage.adoc]:
```suggestion
NOTE: JBOD storage is supported only for Kafka, not for Zookeeper.
```

File: documentation/book/assembly-overview.adoc [hunk 3]
```diff
@@ -21,4 +21,6 @@ This guide describes how to install and use {ProductLongName}.
 
 include::ref-key-features.adoc[leveloffset=+1]
 
+include::con-considerations-for-data-storage.adoc[leveloffset=+1]
```
Comment (scholzj) [inline documentation/book/assembly-overview.adoc]: Is this really the right placement for this chapter? I would probably put it somewhere under 2.3 and 3.1.1.

File: documentation/book/con-considerations-for-data-storage.adoc (hunk 1 above)
Comment (scholzj) [inline documentation/book/con-considerations-for-data-storage.adoc]:
```suggestion
{ProductName} requires block storage and works well with cloud-based block storage solutions, such as Amazon Elastic Block Store (EBS). The use of file storage is not recommended.
```

I'm not a native speaker. But _including Amazon EBS_ sounds to me as if that is not one of the cloud based storage solutions.

File: documentation/book/assembly-overview.adoc (hunk 3 above)
Comment (PaulRMellor) [reply documentation/book/assembly-overview.adoc]: What about we leave it where it is, but link back from sections 2.3 and 3.1.1?

It was placed at the same location as the section in the RHEL doc, which also has a section on supported configurations in the opening chapter. (Maybe we should consider a restructure and the possibility of separating the deployment and operator content into separate guides in the future.)

File: documentation/book/con-considerations-for-data-storage.adoc (hunk 1 above)
Comment (PaulRMellor) [reply documentation/book/con-considerations-for-data-storage.adoc]: Changed

File: documentation/book/assembly-overview.adoc (hunk 3 above)
Comment (scholzj) [reply documentation/book/assembly-overview.adoc]: Well, I'm not sure I reviewed it when it was originally added to RHEL. But to place it there does not make sense at all for me. It feels like it is completely out of place there. People have no clue how it works but we start telling them something about block storage, iSCSI, JBOD etc. It is even before the document conventions.

Comment (PaulRMellor) [reply documentation/book/assembly-overview.adoc]: Thanks Jakub. Let's decide on a new place for 1.1.0. How about:

- RHEL: After '2.3 Installing AMQ Streams'
- OCP: After '2.3 Kafka cluster'

Comment (scholzj) [reply documentation/book/assembly-overview.adoc]: That sounds good to me.

**Pull request merged**